lib/sdam/topology.js

```diff
@@ -101,6 +101,10 @@ class Topology extends EventEmitter {
    * @param {number} [options.localThresholdMS=15] The size of the latency window for selecting among multiple suitable servers
    * @param {number} [options.serverSelectionTimeoutMS=30000] How long to block for server selection before throwing an error
    * @param {number} [options.heartbeatFrequencyMS=10000] The frequency with which topology updates are scheduled
+   * @param {boolean} [options.directConnection] will set topology to Single
+   * @param {string} [options.replicaSet] will set topology to ReplicaSetNoPrimary
```

Comment:
`replicaSet` should be the only supported option, `setName` and `rs_name` can and should be removed in this major revision. Also I think the docstrings for all of these are a bit wanting, `directConnection` means "connect directly to the node without discovering its type" and "replicaSet" is the name of the replica set to connect to (used for validation during initial handshake as well). You can find decent descriptions of these in the [URI Options spec](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst)